manifests/init.pp

```diff
@@ -16,6 +16,8 @@
   $dnskeyname          = undef,
   $pxeserver           = undef,
   $pxefilename         = undef,
+  $ipxe_filename       = undef,
+  $ipxe_bootstrap      = undef,
```

Comment:
I'd prefer some validation to ensures that when `ipxe_filename` is defined that `ipxe_bootstrap` is also defined. This will prevent the manifest from continuing forward and the ipxe declaration silently failing/no-oping when creating the `dhcpd.conf.pxe.erb` template.

The same validation *should* have been there for `pxeserver` and `pxefilename` as well, but are not.